File: src/main/java/no/nav/syfo/controller/domain/tidslinje/Hendelse.java
package no.nav.syfo.controller.domain.tidslinje;

import java.time.LocalDate;
import java.util.HashMap;
import java.util.Map;

public class Hendelse {

    public String id;
    public LocalDate inntruffetdato;
    public Hendelsestype type;
    public Integer antallDager;
    public String tekstkey;
    public Map<String, Object> data = new HashMap<>();

    public Hendelse withInntruffetdato(final LocalDate inntruffetdato) {
        this.inntruffetdato = inntruffetdato;
        return this;
    }

    public Hendelse withType(final Hendelsestype type) {
        this.type = type;
        return this;
    }

    public Hendelse withData(String key, Object data) {
        this.data.put(key, data);
        return this;
    }

    public Hendelse withAntallDager(final Integer antallDager) {
        this.antallDager = antallDager;
        return this;
    }

    public Hendelse withTekstkey(final String tekstkey) {
        this.tekstkey = tekstkey;
        return this;
    }

    public Hendelse withId(final String id) {
        this.id = id;
        return this;
    }
}

File: src/main/kotlin/no/nav/syfo/controller/domain/tidslinje/Hendelse.kt
package no.nav.syfo.controller.domain.tidslinje

import no.nav.syfo.controller.domain.NaermesteLeder
import java.time.LocalDate
import java.util.*

class Hendelse {
    var id: String? = null
    var inntruffetdato: LocalDate? = null
    var type: Hendelsestype? = null
    var antallDager: Int? = null
    var tekstkey: String? = null
    var data: MutableMap<String, Any?> = HashMap()
    fun withInntruffetdato(inntruffetdato: LocalDate?): Hendelse {
        this.inntruffetdato = inntruffetdato
        return this
    }

    fun withType(type: Hendelsestype?): Hendelse {
        this.type = type
        return this
    }

    fun withData(key: String, data: NaermesteLeder?): Hendelse {
        this.data[key] = data
        return this
    }

    fun withAntallDager(antallDager: Int?): Hendelse {
        this.antallDager = antallDager
        return this
    }

    fun withTekstkey(tekstkey: String?): Hendelse {
        this.tekstkey = tekstkey
        return this
    }

    fun withId(id: String?): Hendelse {
        this.id = id
        return this
    }
}